expand-fp: Refactor modification status handling (NFC)

frederik-h opened the pull request:
This is a small refactoring to set the return value of the runImpl function which indicates whether or not the IR has been changed in a single place instead of doing it separately at the insertion of supported instructions into the worklist.

frederik-h committed:
expand-fp: Refactor modification status handling (NFC)

This is a small refactoring to set the return value
of the runImpl function which indicates whether or
not the IR has been changed in a single place instead
of doing it separately at the insertion of supported
instructions into the worklist.

arsenm commented on llvm/lib/CodeGen/ExpandFp.cpp:
```diff
@@ -941,10 +941,9 @@ static void scalarize(Instruction *I,
-      ScalarizedI->copyIRFlags(I, true);
-      Worklist.push_back(ScalarizedI);
-    }
+    Instruction *ScalarizedI = cast<Instruction>(NewOp);
```
I'm not sure removing the dyn_cast is safe, it's there to guard against the case where the IRBuilder constant folded something. I'm surprised there is no test for this case

arsenm replied:
if there aren't tests where each and both operand are constants / constant vectors there should be

frederik-h committed:
[AMDGPU] Add test cases for frem with constant operands

frederik-h committed:
expand-fp: Revert "scalarize" changes

The "dyn_cast" needs to be there as witnessed by the test case in
CodeGen/AMDGPU/frem.ll with two constant vector operands.

Refactor the loop that visits the instructions to allow for a single
assignment to the "Modified" variable.

frederik-h committed:
expand-fp: Replace if-else by switch-case in Worklist processing

This does reflect the structure of the instruction visiting function
and is more readable.

frederik-h committed:
Merge remote-tracking branch 'upstream/main' into expand-fp-refactor-modified

frederik-h committed:
Merge remote-tracking branch 'upstream/main' into expand-fp-refactor-modified

frederik-h committed:
Merge remote-tracking branch 'upstream/main' into expand-fp-refactor-modified

frederik-h committed:
Merge remote-tracking branch 'upstream/main' into expand-fp-refactor-modified
